monero/src/public_key.rs

```diff
@@ -70,11 +84,36 @@ impl <N: MoneroNetwork> MoneroPublicKey<N> {
         self.view_key
     }
 
+    /// Returns the format of the Monero address.
+    pub fn format(&self) -> Format {
+        self.format.clone()
+    }
+
     fn scalar_mul_by_b_compressed(bits: &[u8; 32]) -> [u8; 32] {
         let point = &Scalar::from_bits(*bits) * &ED25519_BASEPOINT_TABLE;
         let compressed = *point.compress().as_bytes();
         compressed
     }
+
+    /// Returns a Monero subaddress public key given a private key and subaddress private view key.
+    pub fn to_subaddress_public_key(
```

Comment:
This method doesn't take "self" and should be in the private key struct. Otherwise, it should be a "from" method.